hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java

```diff
@@ -99,7 +103,16 @@ public static Scope importAndCreateScope(String name, String encodedParent) {
     if (encodedParent != null && encodedParent.length() > 0) {
       StringBuilder builder = new StringBuilder();
       builder.append(encodedParent);
-      parentSpan = tracer.extract(StringCodec.FORMAT, builder);
+      try {
+        parentSpan = tracer.extract(StringCodec.FORMAT, builder);
+      } catch (Exception ex) {
+        if (LOG.isDebugEnabled()) {
+          LOG.debug("Can't extract tracing from the message.", ex);
+        } else {
+          LOG.warn(
```

Comment:
Should we just LOG.warn() here only?

Reply:
My log was full with stack traces in case of a problem. I removed the stack trace from WARN and only the e.getMessage is printed out. I think it should be enough to notice the problem.

But I have no strong opinion. What do you prefer? WARN + full exception all the time?

Reply:
Can we use a  log throttler for this WARN message?

Reply:
Thanks the hint. I was not familiar with `org.apache.hadoop.log.LogThrottingHelper` but I learned it today.

 * Unfortunately -- based on the javadoc -- it's not thread safe. I can't use it in the static class without significant modification.
 * I also noticed that this modification couldn't help at all, as the exception is catched by the ' PropagationRegistry$ExceptionCatchingExtractorDecorator' not this try catch block (see the stack trace in the summary.

It can be simpler to keep the original version of the error handling in this patch. Adopting the throttle or configuring the decorator PropagationRegistry can be bigger task and we can do it later. What ado you think?